core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -443,18 +456,102 @@ private SearchResult parseToSearchResult(@Nonnull final IMAPResponse[] ir) throw
             }
             final List<Long> v = new ArrayList<Long>(); // will always return a non-null array
 
+            Long modSeq = null;
+
             // Grab all SEARCH responses
             long num;
             for (final IMAPResponse sr : ir) {
                 // There *will* be one SEARCH response.
                 if (sr.keyEquals("SEARCH")) {
                     while ((num = sr.readLong()) != -1) {
-                        v.add(Long.valueOf(num));
+                        v.add(num);
+                    }
+                    if (sr.readByte() == '(') {
+                        final String s = sr.readAtom();
+                        if (s.equalsIgnoreCase("MODSEQ")) {
+                            modSeq = sr.readLong();
+                        }
+                    }
+                }
+            }
+
+            return new SearchResult(v, modSeq);
+        }
+
+        /**
+         * Parses the responses from Store command and UID Store command to a @{code StoreResult} object.
+         *
+         * @param ir the list of responses from Store or UID Store command, the input responses array should contain the tagged/final one
+         * @return StoreResult object constructed based on the given IMAPResponse array,
+         * @throws ImapAsyncClientException when tagged response is not OK and NO or given response length is 0
+         */
+        @Nonnull
+        private StoreResult parseToStoreResult(@Nonnull final IMAPResponse[] ir) throws ImapAsyncClientException, IOException, ProtocolException {
```

Comment:
This is ok, but we will need a way to parse fetch extension (non-standard attributes)